src/Detector.php

```diff
@@ -67,19 +67,26 @@ public function determineMimeType(\SplFileObject $file): string
     /**
      * Read width and height from a image file.
      *
-     * @param string $filePath
+     * @param \SplFileObject $file File object
      * @return array [$width, $height]
      * @throws \InvalidArgumentException Unreadable SplFileInfo
      */
-    public function determineDimensions(string $filePath): array
+    public function determineDimensions(\SplFileObject $file): array
     {
-        $stream = new \SplFileInfo($filePath);
-        if (!$stream->isReadable()) {
-            throw new \InvalidArgumentException('File path ' . $filePath . ' is not existed');
+        if (false !== $path = $file->getRealPath()) {
+            $info = getimagesize($path);
+
+            return [$info[0], $info[1]];
+        }
+
+        $file->rewind();
+        $buffer = '';
+        while (false === $file->eof()) {
+            $buffer .= $file->fread(512);
         }
-        list($width, $height) = getimagesize($filePath);
+        $info = getimagesizefromstring($buffer);
```

Comment:
The same as above applies to [`getimagesizefromstring`](https://secure.php.net/manual/en/function.getimagesizefromstring.php): it may return `false`.